src/stores/index.js

```diff
@@ -179,18 +185,25 @@ const Store = Reflux.createStore({
     if (customUrl === '') {
       this._clearForm();
       this._clearConnection();
-    } else if (!Connection.isURI(customUrl)) {
+
+      return;
+    }
+
+    if (!Connection.isURI(customUrl)) {
```

Comment:
Why do we need boot `Connection.isURI(customUrl)` and `Connection.from`? isn't one included in the other?

Reply:
They are different. `Connection.isURI` checks uri prefix for `mongodb://` or `mongodb+srv://` and returns boolean what allows us throw a validation error.

`Connection.from` always returns uri either with users' params or with default values, so the result of `Connection.from` won't be helpful here.